library/gen_x12_837.inc.php

```diff
@@ -38,236 +43,152 @@ function gen_x12_837($pid, $encounter, &$log, $encounter_claim = false)
     "*" . $claim->x12gsisa15() .
     "*:" .
     "~\n";
 
     $out .= "GS" .
-    "*HC" .
+    "*" . "HC" .
     "*" . $claim->x12gsgs02() .
     "*" . trim($claim->x12gs03()) .
     "*" . date('Ymd', $today) .
     "*" . date('Hi', $today) .
-    "*1" .
-    "*X" .
+    "*" . "1" .
+    "*" . "X" .
     "*" . $claim->x12gsversionstring() .
     "~\n";
 
     ++$edicount;
     $out .= "ST" .
-    "*837" .
-    "*0021" .
-    // Spec says the following is optional, so should be able to leave it out.
-    ($CMS_5010 ? ("*" . $claim->x12gsversionstring()) : "") .
+    "*" . "837" .
+    "*" . "0021" .
+    "*" . $claim->x12gsversionstring() .
     "~\n";
 
     ++$edicount;
     $out .= "BHT" .
-    "*0019" .                             // 0019 is required here
-    "*00" .                               // 00 = original transmission
-    "*0123" .                             // reference identification
+    "*" . "0019" .                             // 0019 is required here
+    "*" . "00" .                               // 00 = original transmission
+    "*" . "0123" .                             // reference identification
     "*" . date('Ymd', $today) .           // transaction creation date
     "*" . date('Hi', $today) .            // transaction creation time
-    ($encounter_claim ? "*RP" : "*CH") .  // RP = reporting, CH = chargeable
+    "*" . ($encounter_claim ? "RP" : "CH") .  // RP = reporting, CH = chargeable
     "~\n";
 
-    if (!$CMS_5010) {
-        // This segment was deleted for 5010.
-        ++$edicount;
-        $out .= "REF" .
-        "*87" .
-        "*" . $claim->x12gsversionstring() .
-        "~\n";
-    }
-
     ++$edicount;
-
     if ($claim->federalIdType() == "SY") { // check entity type for NM*102 1 == person, 2 == non-person entity
-        $tempName = $claim->billingFacilityName();
-        $partsName = explode(' ', $tempName);// Loop 1000A submitter entity == person
-        $num_parts = count($partsName);
-        switch ($num_parts) {
-            case "2":
-                $firstName = $partsName[0];
-                $middleName = '';
-                $lastName = $partsName[1];
-                $suffixName = '';
-                break;
-            case "3":
-                $firstName = $partsName[0];
-                $middleName = $partsName[1];
-                $lastName = $partsName[2];
-                $suffixName = '';
-                break;
-            case "4":
-                $firstName = $partsName[0];
-                $middleName = $partsName[1];
-                $lastName = $partsName[2];
-                $suffixName = $partsName[3];
-                break;
-            default:
-                $log .= "*** submitter name in 1000A loop has more than 4 parts, may not be desirable\n";
-                $firstName = $partsName[0];
-                $middleName = $partsName[1];
-                $lastName = $partsName[2];
-                $suffixName = $partsName[3];
```

Comment:
$suffixName now looks like a loose end below. Should we remove this or support it by collecting the provider suffix?